libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -111,6 +111,20 @@ static bool _createKeepAliveOperation( const IotMqttNetworkInfo_t * pNetworkInfo
                                        uint16_t keepAliveSeconds,
                                        _mqttConnection_t * pMqttConnection );
 
+/**
+ * @brief Initialize a network connection, creating it if necessary
+ *
+ * @param[in] pNetworkInfo User-provided network information for the connection
+ * connection.
+ * @param[out] pNetworkConnection On success, the created and/or initialized network connection.
+ * @param[out] pCreatedNetworkConnection On success, `true` if the network connection was created; `false` if it was only initialized.
+ *
+ * @return Any #IotNetworkError_t, as defined by the network stack.
+ */
+static IotNetworkError_t _initializeNetworkConnection( const IotMqttNetworkInfo_t * pNetworkInfo,
```

Comment:
Do you need the pCreatedNetworkConnection output variable?  Can't you determine if a network connection was created based on the return value?

Also nit: This function should be called `_createNetworkConnection`.  We use the terms `_create/_destroy` for objects, while `Init/Cleanup` refer to library initialization sequences (such as IotMqtt_Init/IotMqtt_Cleanup).

Reply:
We can't eliminate the out parameter because this function handles 2 cases; the case in which the connection is created (`pNetworkInfo->createNetworkConnection == true`), _and_ the case in which an existing connection is re-used (`pNetworkInfo->createNetworkConnection == false`).  The out parameter lets the caller know if they need to clean up the connection.

Changed the name of this function to `_createNetworkConnection()`.

Reply:
I think having a `pCreatedNetworkConnection` variable in a function named `_createNetworkConnection` might be a source of confusion since its value isn't tied to the function's success. Perhaps renaming it to something like `pOwnNetworkConnection` instead?

Reply:
After another go-round in `IotMqtt_Connect()` I've been able to eliminate this out parameter.  Agree that it was confusing.